db/db_impl/db_impl_merge.cc

```diff
@@ -202,25 +195,12 @@ Status DBImpl::MergeDisjointInstances(const MergeInstanceOptions& merge_options,
 
       if (merge_options.merge_memtable) {
         if (!cfd->mem()->IsEmpty()) {
-          // Freeze memtable. Only immutable memtables can be shared.
-          auto& cf_mopts = *cfd->GetLatestMutableCFOptions();
-          auto* new_mem = cfd->ConstructNewMemtable(cf_mopts, max_seq_number);
-          new_mem->Ref();
-          cfd->imm()->Add(cfd->mem(), &to_delete);
-#ifndef ROCKSDB_LITE
-          MemTableInfo memtable_info;
-          memtable_info.cf_name = cf_name;
-          memtable_info.first_seqno = cfd->mem()->GetFirstSequenceNumber();
-          memtable_info.earliest_seqno =
-              cfd->mem()->GetEarliestSequenceNumber();
-          memtable_info.num_entries = cfd->mem()->num_entries();
-          memtable_info.num_deletes = cfd->mem()->num_deletes();
-          db->NotifyOnMemTableSealed(cfd, memtable_info);
-#endif  // ROCKSDB_LITE
-          cfd->SetMemtable(new_mem);
-          SuperVersionContext sv_context(/* create_superversion */ true);
-          db->InstallSuperVersionAndScheduleWork(cfd, &sv_context, cf_mopts);
-          sv_context.Clean();
+          WriteContext write_context;
+          assert(log_empty_);
+          s = SwitchMemtable(cfd, &write_context);
```

Comment:
Is it okay that new memtable is not with `max_seq_number`

Reply:
LastSequence is set with max_seq_number above.

Reply:
Seems only

Reply:
Oh, then it's not needed, the source DBs' sequence number are unchanged.